### PR TITLE
Migrated deprecated TSlint to ESlint

### PR DESCRIPTION
**Pull request by saneDG**
Migrated deprecated linter. Migration made by following [this](https://code.visualstudio.com/api/advanced-topics/tslint-eslint-migration) guide.

- Add following packages as dev-dependency:
  `eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin`
- Remove tslint dependency
- Remove tslint-config
- Migrate tslint-config to eslint-config (tslint-to-eslint-config)
- modify package.json to run ESlint on commit (script: "lint-staged")

### CODE REVIEW

**Comment by saneDG** [comment]
Please test this configuration on your machine before approve 🙇

File: packages/frontend/package.json [hunk 1]
```diff
@@ -16,6 +16,10 @@
     "web-vitals": "^2.1.4"
   },
   "devDependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.38.0",
```
Comment (NinaDang97) [inline packages/frontend/package.json:19]: Hey I see some duplicates in frontend and backend. Could you reinstall in root these dependencies?

File: package.json [hunk 2]
```diff
@@ -30,18 +30,14 @@
...
-    "tslint-react": "^5.0.0"
+    "prettier": "^2.7.1"
   },
   "lint-staged": {
     "*.{js,json,jsx,css,md}": [
       "prettier --write"
     ],
     "*.{ts,tsx}": [
-      "tslint --fix",
+      "eslint --fix",
```
Comment (NinaDang97) [inline package.json:40]: When commit, I got this error:
Screenshot 2022-09-26 at 14 17 17

**Review by NinaDang97 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (saneDG) [reply package.json:40]: Just noticed same error. I'll try to fix that.

**Comment by mehtis** [comment]
Note unrelated to the commit itself: Please follow the PR naming convention described in the README. With the most important bit being having the ticket number, so that this PR can be linked to it/found later on.

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    'prettier',
+  ],
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    project: 'tsconfig.json',
+    sourceType: 'module',
+  },
+  plugins: [
```
Comment (NinaDang97) [inline .eslintrc.js:28]: Maybe you install all these plugins as devDependencies? `npx eslint .` works for me after the install.
Screenshot 2022-09-27 at 13 01 34

**Pull request closed without merging**